Automattic/wp-calypso
# Global Styles on Personal A/B: Prepare hook

Fixes https://github.com/Automattic/dotcom-forge/issues/2898

## Proposed Changes

* Updates the `useSiteGlobalStylesStatus` hook so it returns a new `globalStylesInPersonalPlan` prop which indicates whether Global Styles are available in the Personal plan due to the upcoming A/B test.
* The new prop is fetched from the GS status endpoint rather that using the ExPlat library so we don't need to deal with users who are not site owners.
* If the site has not been created yet, we fallback to the ExPlat endpoint. We don't use the ExPlat library because it is not supported on SSR pages such as the logged-out theme showcase.
* The `usePremiumGlobalStyles` hook has been removed in favor of `useSiteGlobalStylesStatus`,  since they both were equivalent.

## Testing Instructions

- Apply these changes locally.
- Pick any of the current usages of `useSiteGlobalStylesStatus` and observes the `globalStylesInPersonalPlan` prop:

```diff
diff --git a/client/components/theme/index.jsx b/client/components/theme/index.jsx
index 4b53388591..0139a5fa78 100644
--- a/client/components/theme/index.jsx
+++ b/client/components/theme/index.jsx
@@ -352,6 +352,7 @@ const ConnectedTheme = connect(
 )( localize( Theme ) );
 
 export default ( props ) => {
-       const { shouldLimitGlobalStyles } = useSiteGlobalStylesStatus( props.siteId );
+       const { globalStylesInPersonalPlan, shouldLimitGlobalStyles } = useSiteGlobalStylesStatus( props.siteId );
+       console.log( globalStylesInPersonalPlan );
        return <ConnectedTheme { ...props } shouldLimitGlobalStyles={ shouldLimitGlobalStyles } />;
 };
```
- Assign yourself to the `control` variant.
- Remove the cached assignment from your sandbox: `wp_cache_delete( 'global-styles-on-personal-<BLOG_ID>', 'a8c_experiments' );`
- Wait a minute for the new assignment to be propagated across ExPlat.
- Make sure `globalStylesInPersonalPlan` is `false`.
- Assign yourself to the `treatment` variant.
- Remove the cached assignment again.
- Wait another minute.
- Make sure `globalStylesInPersonalPlan` is `true`.
